jdat_notebooks/preimaging/preimaging_02_calwebb.ipynb

```diff
@@ -0,0 +1,282 @@
+    "outfile = outlist \n",
+    "\n",
+    "\n",
+    "for k, filename in enumerate(files):\n",
+    "    Detector1Pipeline.call(filename, config_file=calwebb_detector1_file, \n",
+    "                           output_file=outfile[k]);\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we run the second stage of the pipeline."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# gather the F150W yaml files back in the working directory\n",
+    "cwd = os.getcwd()\n",
+    "filter_pattern = os.path.join(cwd, 'yaml', filtname.lower(), 'jw*yaml') \n",
+    "files = glob(filter_pattern)[:] # Use [0:2] to test the first two files.\n",
+    "inputyaml = files\n",
+    "namelist = []\n",
+    "outlist = []"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Executing calwebb_image2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We select the *_rate.fits files and execute calwebb_image2 to generate the *_cal.fits files."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for input_yaml_path in inputyaml:\n",
+    "    print(input_yaml_path)\n",
+    "    base = os.path.basename(input_yaml_path)\n",
+    "    name = os.path.splitext(base)[0]\n",
+    "   \n",
+    "    exi = os.path.exists(name + '_cal.fits')\n",
+    "    if exi == True:\n",
+    "        continue\n",
+    "\n",
+    "    namelist.append(name + '_rate.fits')\n",
+    "    outlist.append(name + '.fits')\n",
+    "        \n",
+    "print(namelist)\n",
+    "    \n",
+    "files = namelist \n",
+    "outfile = outlist "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Generate the _cal.fits files."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Select configuration file\n",
+    "calwebb_image2_file = os.path.join('preimaging', 'calwebb_image2.cfg')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for k, filename in enumerate(files):\n",
+    "    print('-'*60)\n",
+    "    print(filename)\n",
+    "\n",
+    "    #here we assign the distortion file according to the detector\n",
+    "    chip = [w for w in filename.split('_') if 'nrc' in w.lower()][0]  \n",
+    "    wcs_reffile = '{}_FULL_from_pysiaf_distortion.asdf'.format(chip.upper())\n",
+    "    wcs_reffile = os.path.join('preimaging', wcs_reffile)\n",
+    "    \n",
+    "    m  =  calwebb_image2.Image2Pipeline(config_file=calwebb_image2_file, \n",
+    "                                        output_file=outfile[k])\n",
+    "    m.assign_wcs.override_distortion = wcs_reffile\n",
+    "    m.run(filename)\n",
```

Comment:
When using the run() method to call a pipeline, the configuration file is ignored and the pipeline is run with default settings (or whatever overrides you provide). https://jwst-pipeline.readthedocs.io/en/latest/jwst/stpipe/user_step.html#running-a-step-in-python